packages/matrix/helpers/index.ts

```diff
@@ -12,6 +14,46 @@ interface LoginOptions {
   expectFailure?: true;
 }
 
+export const test = base.extend<{ synapse: SynapseInstance }>({
+  synapse: async ({}, use) => {
+    let synapseInstance = await synapseStart();
+    await use(synapseInstance);
+    await synapseStop(synapseInstance.synapseId);
+  },
+
+  page: async ({ page, synapse }, use) => {
+    // Setup overrides
+    await setupMatrixOverride(page, synapse);
+    await use(page);
+  },
+});
+
+export async function setupMatrixOverride(
+  page: Page,
+  synapse: SynapseInstance,
+) {
+  // Save the original goto function
+  const originalGoto = page.goto.bind(page);
+
+  // Patch the goto function
+  page.goto = async (url, options) => {
+    const newUrl = new URL(url);
+    const params = new URLSearchParams(newUrl.search);
+
+    // Override the matrix URL
+    params.set('matrixURL', `http://localhost:${synapse.mappedPort}`);
+    newUrl.search = params.toString();
+
+    // Call the original goto function with the new URL
+    return originalGoto(newUrl.toString(), options);
```

Comment:
Missing await here. Please try to be careful around missing async, it leads to flaky tests and super hard to debug.